src/services/connect.ts

```diff
@@ -0,0 +1,36 @@
+import { StellarSDK } from '../api/stellarSdk';
+import type { Keypair } from 'stellar-sdk';
+
+const encryptKey = (key: string, salt: number) => {
+    const encodedData = btoa(key);
+    const saltValue = Math.floor(Math.random() * 10000000 * salt);
+    return encodedData + saltValue;
+};
+
+const saveSecretKeyInSessionStorage = (key: string) => {
+    const encodedKey = encryptKey(key, 1000);
+    sessionStorage.setItem('key', encodedKey);
+};
+
+const connectWithSecretKey = (key: string): Keypair | void => {
+    try {
+        const sourceKeys = StellarSDK.Keypair.fromSecret(key);
+        saveSecretKeyInSessionStorage(key);
+        return sourceKeys;
+    } catch (e) {
+        return console.error('Invalid key: ', e);
```

Comment:
you should re-throw the error and therefore this function should never return `void`